File: 14-08/maiorValorMatriz.py

# Recebendo numero de linhas e colunas da matriz
nLinhas = int(input("Digite o numero de linhas da matriz: "))
nColunas = int(input("Digite o numero de colunas da matriz:"))

# Declarando matriz
matriz = []

# Entrada de dados da matriz
for i in range(nLinhas):
    linha = []
    for j in range(nColunas):
        valor = float(input(f"Digite o valor [{i}] [{j}] da matriz: "))
        linha.append(valor)
    matriz.append(linha)

# Calculo maior valor dentro da matriz
vmaior = 0
for i in range(nLinhas):
    for j in range(nColunas):
        if matriz[i][j] > vmaior:
            vmaior = matriz[i][j]

# Exibição da matriz e seu maior valor
print("Matriz: ")
for linha in matriz:
    print(linha)

print(f"O maior valor é: {vmaior}") 

File: 14-08/maiorValorMatriz.c


#include <stdio.h>
int main(){

	// Declaração de variaveis
	int nLinhas1, nColunas1;
	
	// Entrada do numero de linhas e de colunas da matriz
	printf("Digite o numero de linhas da matriz: ");
	scanf("%d", &nLinhas1);
	printf("Digite o numero de colunas da matriz: ");
	scanf("%d", &nColunas1);
	
    // Declaração da matriz
	double matriz1[nLinhas1][nColunas1];
    double vmaior;
    
    // Entrada de dados da matriz
	for(int i = 0; i<nLinhas1; i++){
		for(int j = 0; j<nColunas1;j++){
			printf("Digite o elemento [%d][%d] da matriz1: ", i, j);
			scanf("%lf", &matriz1[i][j]);
		}
	}

    // Calculo maior valor 
    vmaior = 0;
    for(int i = 0; i<nLinhas1; i++){
        for(int j = 0; j<nColunas1; j++){
            if (matriz1[i][j] > vmaior){
                vmaior = matriz1[i][j];
            }
        }
    }
    // Exibição da matriz e do maior valor
    printf("Matriz: \n");
    for(int i = 0; i < nLinhas1; i++){
        for(int j = 0; j < nColunas1; j++){
            printf("%.2lf ", matriz1[i][j]);
        }
    printf("\n");
    }
    printf("O maior valor é: %.2lf\n", vmaior);
}



